CONTRIBUTING.md

```diff
@@ -0,0 +1,130 @@
+
+## Contributing
+
+### Contributor
+
+We are very happy that you consider implementing algorithms and data structures for others! This repository is
+referenced and used by learners from around the globe. Being one of our contributors, you agree and confirm that:
+
+- You did your work - plagiarism is not allowed.
+  - Any plagiarized work will not be merged.
+- Your work will be distributed under [MIT License](LICENSE) once your pull request is merged.
+- Your submitted work must fulfill our styles and standards.
+
+**New implementations** are welcome! For example, new solutions to a problem, different representations of a graph data
+structure, or algorithm designs with different complexity.
+
+**Improving comments** and **writing proper tests** are also highly welcome.
+
+### Contribution
+
+We appreciate any contribution, from fixing grammar mistakes to implementing complex algorithms. Please read this
+section if you are contributing to your work.
+
+If you submit a pull request that resolves an open issue, please help us to keep our issue list small by adding
+`closes: #{$ISSUE_NO}` to your commit message. GitHub will use this tag to auto-close the issue if your PR is merged.
+
+#### What is an Algorithm?
+
+An Algorithm is one or more functions (or classes) that:
+
+- take one or more inputs.
+- perform some internal calculations or data manipulations.
+- return one or more outputs.
+- have minimal side effects.
+
+Algorithms should be packaged in a way that would make it easy for readers to put them into larger programs.
+
+Algorithms should:
+
+- have intuitive class and function names that make their purpose clear to readers.
+- use TypScript naming conventions and intuitive variable names to ease comprehension.
+- be flexible to take different input values.
+- raise TypeScript exceptions (RangeError, etc.) on erroneous input values.
+
+Algorithms in this repo should not be how-to examples for existing TypeScript packages. Instead, they should perform
+internal calculations or manipulations to convert input values into different output values. Those calculations or
+manipulations can use data types, classes, or functions of existing TypeScript packages but each algorithm in this repo
+should add unique value.
+
+#### File Naming Convention
+
+- Filenames should use the UpperCamelCase (PascalCase) style.
+- There should be no spaces in filenames.
+- **Example:** `UserProfile.ts` is allowed. Do not use `userprofile.ts`, `Userprofile.ts`, `user-Profile.ts`, `userProfile.ts`; these methods are discouraged.
+
+#### Commit Messages Formatting
+
+- Prefer to use the following format: `<type>: <short summary>`. If necessary, put any extra information in the description.
+- Commit types include (but are not limited to): 
+  - **docs**: Documentantion only changes
+  - **feat**: A new feature
+  - **fix**: A bug fix
+  - **chore**: Miscellaneous stuff that does not match any of the above
```

Comment:
I wouldn't use chore for "anything else", especially as this doesn't cover tests yet; would adding tests be a "feat"? "chore" is for CI / code quality / minor quality of life improvements like fixing typos etc. which doesn't impact the way the code works or is tested IMO.